illumina_uploader/illumina_uploader.py

```diff
@@ -119,65 +122,99 @@ def main():
+                            runArgs["inFile"] = folderToUpload
+                            #Mail send before start
+                            status = "STARTED"
+                            reason = ""
+                            if emailInfo['enabled']:
+                                subject = emailInfo["mailsubject"].format(status=status)
+                                body = emailInfo["mailbody"].format(folderToUpload=folderToUpload, status=status, timeOfMail=getDateTimeNow(), reason=reason)
+                                if ploverEmailInfo["enabled"]:
+                                    sendEmailUsingPlover(ploverEmailInfo["emailurl"], { **ploverEmailArgs, 'subject': subject, 'body': body}, logger)
+                                if mcmsEmailInfo["enabled"]:
+                                    mcms.send_email(mcmsEmailInfo['emailurl'], mcmsEmailInfo['mailto'], subject, body) 
+                            runArgs["runscache"] = runsCache
+                            isSuccessful = False
+                            runArgs["starttime"] = getDateTimeNowIso()
+                            try:
+                                isSuccessful = uploadRunToServer(context, runArgs)
+                            except (socket.error, OSError) as error:
+                                logger.error("Fatal OS / network error: {0}".format(error))
+                                reason = "Fatal OS / network error.. will try again in {0} minutes".format(localInfo["sleeptime"])
+                            status = "FINISHED" if isSuccessful else "FAILED"
+                            logger.info("Marking in DB as {0}".format(status))
+                            dbObject.markFileInDb(folderToUpload, status)
+                            #Create json file
+                            if isSuccessful:
+                                scpUploadCompleteJson(context, runArgs)
+                            #Mail send after done, update subject and body
+                            if emailInfo['enabled']:
+                                subject = emailInfo["mailsubject"].format(status=status)
+                                body = emailInfo["mailbody"].format(folderToUpload=folderToUpload, status=status, timeOfMail=getDateTimeNow(), reason=reason)
+                                if ploverEmailInfo["enabled"]:
+                                    sendEmailUsingPlover(ploverEmailInfo["emailurl"], { **ploverEmailArgs, 'subject': subject, 'body': body}, logger)
+                                if mcmsEmailInfo["enabled"]:
+                                    mcms.send_email(mcmsEmailInfo['emailurl'], mcmsEmailInfo['mailto'], subject, body) 
+                    #Goto sleep (displayed in minutes)
+                    logger.info("Sleeping for {0} minutes".format(localInfo["sleeptime"]))
+                    
+                    time.sleep(sleeptimeInSeconds)
+                except FileNotFoundError as error:
```

Comment:
We want the script to continue running even if there's a file/folder/network error. Let it try again after some time.